src/lib/parser.c

```diff
@@ -105,90 +155,234 @@ parser_error_t parser_getItem(const parser_context_t *ctx,
+                    break;
+                }
+                case FIELD_PROPOSAL_ID: { //Proposal Id
+                    snprintf(outKey, outKeyLen, "ProposalId");
+                    uint8_t bcdOut[20]; //Must be  at most outValueLen/2
+                    uint16_t bcdOutLen = sizeof(bcdOut);
+                    bignumBigEndian_to_bcd(bcdOut, bcdOutLen, parser_tx_obj.votemsg.proposalIdPtr,
+                                           parser_tx_obj.votemsg.proposalIdLen);
+                    bignumBigEndian_bcdprint(outValue, outValueLen, bcdOut, bcdOutLen);
+                    break;
+                }
+                case FIELD_SELECTION: { // Vote option
+                    char *sel;
+                    switch (parser_tx_obj.votemsg.voteOption) {
+                        case VOTE_OPTION_YES:
+                            sel = VOTE_OPTION_YES_STR;
+                            break;
+                        case VOTE_OPTION_NO:
+                            sel = VOTE_OPTION_NO_STR;
+                            break;
+                        case VOTE_OPTION_ABSTAIN:
+                            sel = VOTE_OPTION_ABSTAIN_STR;
+                            break;
+                        default:
+                            sel = VOTE_OPTION_INVALID_STR;
+                    }
+                    MEMCPY(outValue, sel, sizeof(sel));
+                    snprintf(outKey, outKeyLen, "Selection");
+                    break;
+                }
+                default: {
+                    // Handle variable fields
+                    if (displayIdx >= parser_getNumItems(ctx)) {
+                        *pageCount = 0;
+                        return parser_no_data;
+                    }
+                }
+            }
             break;
-        case FIELD_DESTINATION:     // Destination
-            snprintf(outKey, outKeyLen, "Dest");
+        }
+        case Msg_Update: {
+            switch (parser_mapDisplayIdx(ctx, displayIdx))
+            {
+                case FIELD_CHAINID:     // ChainID
+                    snprintf(outKey, outKeyLen, "ChainID");
+                    err = parser_arrayToString(outValue, outValueLen,
+                                               parser_tx_obj.chainID,
+                                               parser_tx_obj.chainIDLen,
+                                               pageIdx, pageCount);
+                    break;
+                case FIELD_CONTRACT_ID: //Contract Id
+                    snprintf(outKey, outKeyLen, "ContractId");
+                    uint8_t bcdOut[20]; //Must be  at most outValueLen/2
+                    uint16_t bcdOutLen = sizeof(bcdOut);
+                    bignumBigEndian_to_bcd(bcdOut, bcdOutLen, parser_tx_obj.updatemsg.contractIdPtr,
+                                           parser_tx_obj.updatemsg.contractIdLen);
+                    bignumBigEndian_bcdprint(outValue, outValueLen, bcdOut, bcdOutLen);
```

Comment:
we need to check if it was actually possible to write the complete value out to both the bcdOutLen and outKey.. otherwise, it would be possible to send a value that is truncated in the output.
As far as I remember, `bcdOutLen` will be zero if the answer didn't fit.. Please check it

Reply:
`bignumBigEndian_bcdprint()` returns bool_true if success, I can grab that and return parse_error if it failed.